packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -300,6 +300,60 @@ function warnIfInvokedDuringConstruction(vm: VM, methodOrPropName: string) {
     }
 }
 
+// List of properties on ElementInternals that are formAssociated can be found in the spec:
+// https://html.spec.whatwg.org/multipage/custom-elements.html#form-associated-custom-elements
+const formAssociatedProps = new Set([
+    'setFormValue',
+    'form',
+    'setValidity',
+    'willValidate',
+    'validity',
+    'validationMessage',
+    'checkValidity',
+    'reportValidity',
+    'labels',
+]);
+
+// Wrap all ElementInternal objects in a proxy to prevent form association when `formAssociated` is not set on an LWC.
+// This is needed because the 1UpgradeableConstructor1 always sets `formAssociated=true`, which means all
+// ElementInternal objects will have form-associated properties set when an LWC is placed in a form.
+// We are doing this to guard against customers taking a dependency on form elements being associated to ElementInternals
+// when 'formAssociated' has not been set on the LWC.
+function createElementInternalsProxy(
+    elementInternals: ElementInternals,
+    isFormAssociated: boolean
+) {
+    const elementInternalsProxy = new Proxy(elementInternals, {
+        set(target, propertyKey, newValue) {
+            const key = typeof propertyKey === 'symbol' ? propertyKey.description : propertyKey;
```

Comment:
In what cases do we get a symbol here?

Reply:
I think I may have overthought this part, technically the second argument passed to the getter can be a `symbol` or `string` and I was trying to guard against the chance that it was a symbol.

Now that I look at the implementation for `ElementInternals` in Chrome, Firefox, and Safari it looks like they're always strings.

Made a few updates to use strings instead!